### ISSUE TITLE
docker-publish.yml doesn't work with uppercase usernames/repositories

### ISSUE THREAD

**Issue by mattmoor** [actions/starter-workflows#1293]
Creating an issue to track the problem highlighted by @bishal-pdMSFT around casing here: https://github.com/actions/starter-workflows/pull/1255

tl;dr `docker push` automatically lowercases repository names when pushing, and cosign rejects the uppercase latters.

I'm pursuing a patch to google/go-containerregistry that allows uppercase repositories when "weak validation" is enabled (the default), which we can try to pull into a patch release of cosign.

However, more tactically we could land a patch to this workflow that just lowercases the repository name for now?

We'll need a change to the action either way, and we'll want the former fix regardless, but I'm happy to put together the latter if we think a lot of folks with uppercase usernames will hit this?

**Comment by imjasonh** [comment]
> tl;dr `docker push` automatically lowercases repository names when pushing, and cosign rejects the uppercase latters.

It looks like `docker push` isn't lowercasing, it's `docker/metadata-action`'s output `tags` that emit a lowercased repo, which is passed into `docker/build-and-push` (see [here](https://github.com/bishal-pdMSFT/docker-app/runs/4434089694?check_suite_focus=true#step:6:29))

It looks like another way to fix this without changing `cosign` would be to have the action call:

```
cosign sign ${{ steps.meta.outputs.tags }}@${{ steps.build-and-push.outputs.digest }}
```

This would end up signing, e.g., `ghcr.io/bishal-pdmsft/docker-app:master@sha256:ec9994d1d8cf472600bbc9fa73d2e6afe865beef000185659f6d0e8f774b47f8` which, despite looking weird, is a valid image ref, which `cosign sign` should accept.

**Comment by mattmoor** [comment]
I confirmed that works here: https://github.com/mattmoor/zero-friction-actions/runs/4435501291?check_suite_focus=true

I'll send a PR shortly.
